tests/base/src/test/java/org/keycloak/tests/admin/AdminConsoleWhoAmILocaleTest.java

```diff
@@ -1,304 +1,352 @@
-            }
+    public void testLocaleRealmTokenAdminNotAllowed() throws Exception {
+        AccessTokenResponse response = accessToken(oAuthClientLocaleOn, ADMIN_CLI_NOT_ALLOWED, SECRET, USER_WITH_LOCALE, PASSWORD);
+
+        Assertions.assertNotNull(response.getAccessToken());
+
+        HttpGet httpGet = new HttpGet(whoAmiUrl(managedRealmOn));
+        httpGet.addHeader("Accept", "application/json");
+        httpGet.addHeader("Authorization", "Bearer " + response.getAccessToken());
+        HttpResponse httpGetResponse = client.execute(httpGet);
+
+        Assertions.assertEquals(Response.Status.FORBIDDEN.getStatusCode(), httpGetResponse.getStatusLine().getStatusCode());
+    }
+
+    private void updateMasterAdminRole() {
+        RoleRepresentation roleRep = masterRealm.admin().roles().get("admin").toRepresentation();
+        masterRealm.admin().users().get(masterAdmin.getId()).roles().realmLevel().add(List.of(roleRep));
+    }
+
+    private AccessTokenResponse accessToken(OAuthClient oAuth, String clientId, String clientSecret, String username, String password) {
+        return oAuth.client(clientId, clientSecret).doPasswordGrantRequest(username, password);
+    }
+
+    private JsonNode getHttpJsonResponse(String url, AccessTokenResponse response, Header... headers) throws IOException{
+        HttpGet httpGet = new HttpGet(url);
+        httpGet.addHeader("Accept", "application/json");
+        httpGet.addHeader("Authorization", "Bearer " + response.getAccessToken());
+
+        for (Header header : headers) {
+            httpGet.addHeader(header.getName(), header.getValue());
         }
+
+        return new ObjectMapper().readTree(client.execute(httpGet).getEntity().getContent());
     }
 
-    @Test
-    public void testLocaleRealmTokenForOtherClientButAllowed() throws Exception {
-        try (ClientAttributeUpdater updater = ClientAttributeUpdater.forClient(adminClient, REALM_I18N_ON, Constants.ADMIN_CLI_CLIENT_ID)
-                .setAttribute(Constants.SECURITY_ADMIN_CONSOLE_ATTR, Boolean.TRUE.toString())
-                .update();
-                Keycloak adminCliClient = AdminClientUtil.createAdminClient(true, REALM_I18N_ON,
-                USER_WITH_LOCALE, PASSWORD, Constants.ADMIN_CLI_CLIENT_ID, null)) {
-            AccessTokenResponse accessToken = adminCliClient.tokenManager().getAccessToken();
-            Assert.assertNotNull(accessToken);
-            String token = accessToken.getToken();
-            JsonNode whoAmI = SimpleHttpDefault
-                    .doGet(whoAmiUrl(REALM_I18N_ON), client)
-                    .header("Accept", "application/json")
-                    .auth(token)
-                    .asJson();
-            Assert.assertEquals(REALM_I18N_ON, whoAmI.get("realm").asText());
-            Assert.assertEquals(USER_LOCALE, whoAmI.get("locale").asText());
-            checkRealmAccess(REALM_I18N_ON, whoAmI);
```

Comment:
So the two tests `testLocaleRealmTokenForOtherClient` and `testLocaleRealmTokenForOtherClientButAllowed` were removed because of some weird admin client inter-working of the new framework?

Reply:
yes, I've replaced it by single test, since the previous tests already has a client with the security.admin.console enabled parameter, this new test is testing when this parameter is null